tests/src/Unit/SocialApiTest.php

```diff
@@ -0,0 +1,560 @@
+use Drupal\Core\Cache\Cache;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Drupal\Core\Config\StorageInterface;
+use Drupal\Core\Config\TypedConfigManagerInterface;
+use Drupal\social_api\Settings\SettingsInterface;
+//Plugin
+use Drupal\social_api\Plugin\NetworkInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Plugin\PluginBase;
+use Drupal\social_api\Plugin\NetworkBase;
+
+
+
+class SocialApiTest extends UnitTestCase {
+  /**
+   * Checks the library required by an implementer.
+   *
+   * @param string $machine_name
+   *   The module machine name.
+   *
+   * @param string $name
+   *   The module name.
+   * @param string $library
+   *   The library machine name.
+   * @param float $min_version
+   *   The min version required.
+   * @param float $max_version
+   *   The max version required.
+   *
+   * @return array
+   *   Requirements messages.
+   */
+
+  public $machine_name;
+  public $name;
+  public $library;
+  public $min_version;
+  public $max_version;
+  protected $config;
+  private $networkManager;
+  protected $form = array();
+  protected $namespaces;
+  protected $module_handler;
+  protected $cache_backend;
+  public $type = "";
+  public $container;
+  private $networkManagers;
+  protected $client;
+  protected $session;
+  protected $sessionPrefix;
+  protected $storage;
+  protected $typed_config;
+  protected $event_dispatcher;
+  //UserManager
+  protected $entityType;
+  protected $entity_type_manager;
+  protected $messenger;
+  protected $logger_factory;
+  protected $plugin_id;
+  protected $user_manager;
+  protected $data_handler;
+  protected $current_user;
+  protected $dataHandler;
+  protected $sessionKeys;
+  //Netowrk
+  protected $configuration;
+  protected $plugin_definition;
+  protected $sdk = 'drupal123';
+
+
+
+  /**
+   * __construct function
+   */
+  public function __construct() {
+       parent::__construct();
+   }
+
+
+  /**
+   * {@inheritdoc}
+   */
+
+  public function setUp() {
+    // enable any other required module
+    parent::setUp();
+    // $this->socialPostEntityDeleteForm = $this->getMock($SocialPostEntityDeleteForm::class, ['getRedirectUrl']);
+  }
+
+  /**
+   * test for class OAuth2Manager
+   */
+
+  public function testOAuth2Manager () {
+    // assertion to check if the file exists
+    // $this->assertFileExists('../drupal8/modules/social_api/src/AuthManager/OAuth2Manager.php');
+
+    $abstractClass = 'Drupal\social_api\AuthManager\OAuth2Manager';
+
+    //
+    // checking for correct getClient and setClient methods
+    //
+     $mockAbstractClass = $this->getMockBuilder($abstractClass)
+       // ->setMethods(array('setClient','getClient'))
+                               ->getMockForAbstractClass();
+       // $mock->expects($this->once())
+       //          ->method('setClient')
+       //          ->with($this->equalTo('something'));
+
+     $mockAbstractClass->setClient('12345');
```

Comment:
`setClient()` returns `$this`. Check that's the case

Reply:
$this will return an object with a string that is passed.